integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -16,30 +16,31 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 import { expect } from "chai";
+import { KJUR } from "jsrsasign";
+import Realm, { ProviderType, UserState } from "realm";
+
 import { importAppBefore } from "../../hooks";
 import {
   randomNonVerifiableEmail,
   randomPendingVerificationEmail,
   randomVerifiableEmail,
 } from "../../utils/generators";
-import { KJUR } from "jsrsasign";
-import Realm, { UserState } from "realm";
-
 import { buildAppConfig } from "../../utils/build-app-config";
 
 type AnyApp = Realm.App<any, any>;
 type AnyUser = Realm.User<any, any, any>;
 
-function expectIsUser(user: Realm.User) {
-  expect(user).to.not.be.undefined;
-  expect(user).to.not.be.null;
-  expect(typeof user).equals("object");
-  expect(typeof user.accessToken).equals("string");
-  expect(typeof user.refreshToken).equals("string");
-  expect(typeof user.id).equals("string");
-  expect(typeof user.identities).equals("object");
-  expect(typeof user.customData).equals("object");
+function expectIsUser(user: Realm.User, providerType?: ProviderType) {
+  expect(user).to.be.an("object");
+  expect(user.accessToken).to.be.a("string");
+  expect(user.refreshToken).to.be.a("string");
+  expect(user.id).to.be.a("string");
+  expect(user.customData).to.be.an("object");
   expect(user).instanceOf(Realm.User);
+  expect(user.identities).to.be.an("array");
+  if (providerType) {
+    expect(user.identities.some((identity) => identity.providerType === providerType)).to.be.true;
+  }
```

Comment:
We could also choose to move this to a separate function to keep the `expectIsUser` focused on type checking.

Reply:
I'd probably inline this in the test function instead 👍

Reply:
This is better IMO 👍